intune-remediations/uninstall-classic-teams/uninstall-machine-wide-installer/detection.ps1

```diff
@@ -0,0 +1,86 @@
+<#
+.SYNOPSIS
+Detects if the Teams Machine-Wide Installer is installed.
+.DESCRIPTION
+Detects if the Teams Machine-Wide Installer is installed by searching the registry for the application name. If the application is found, remediation is required.
+This will be performed by the remediation script and is to be used in an Intune remediation.
+.EXAMPLE
+.\detection.ps1
+Searches the registry for the Teams Machine-Wide Installer application.
+.NOTES
+File Name      : detection.ps1
+Author         : Nicholas Tabb
+Date           : 09/06/2024
+Context        : Computer (System)
+#>
+
+$requireNewTeams = $true # Set to $true to check for the existence of new Teams. This means that remediation will not be required if new Teams is not present.
+$applicationName = "Teams Machine-Wide Installer"
+
+
+# ----- do not modify below this line (unless you know what you are doing) ----- #
+function Get-RegistryKey {
+    <#
+    .SYNOPSIS
+    Retrieves the registry key for the specified application.
+    .DESCRIPTION
+    Retrieves the registry key for the specified application. The function searches the registry hives for the application name.
+    .PARAMETER applicationName
+    The name of the application to search for.
+    .EXAMPLE
+    Get-RegistryKey -applicationName "Teams Machine-Wide Installer"
+    Searches the registry for the Teams Machine-Wide Installer application.
+    #>
+    param (
+        [Parameter(Mandatory = $true, ValueFromPipeline = $true)][string]$applicationName
+    )
+
+    begin {
+        $hives = @("HKLM:\SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\*", 
+            "HKLM:\SOFTWARE\WOW6432Node\Microsoft\Windows\CurrentVersion\Uninstall\*")
+        $keys = @()
+    }   
+
+    process {
+        foreach ($hive in $hives) {
```

Comment:
Instead of defining keys and then appending it like this, you could aggregate as such:

$keys = $hives | ForEach-Object { 
    Get-ItemProperty $_ | Where-Object { $_.DisplayName -Like "$applicationName" }
}

Reply:
Tested further on a couple dozen of my machines with no issues. I also tried implementing my recommendation but there was an oversight with how it works, it will overwrite to the last one in the loop. Either way, looks good!